lib/ace/mode/latex_parser.js

```diff
@@ -33,26 +52,58 @@ var LatexParser = exports.LatexParser = function(bgTokenizer) {
 
                 switch(token.type) {
-                    case "text": case "comment":
-                        var words = this.splitTextToWords(token.value, row, column);
-                        for (var i = 0, wordsLength = words.length; i < wordsLength; ++i) {
-                            if (!dictionary.check(words[i].value)) {
-                                this.myErrors.push({
-                                    row: words[i].row,
-                                    column: words[i].column,
-                                    text: "grammar error",
-                                    type: "error",
-                                    raw: words[i].value
-                                });
-                            }
-                        }
+                    case "comment":
+                        doGetComments && parseRowForTextComments.call(this, token.value, row);
+                        doSpellcheck && parseRowForSpellcheck.call(this, token.value, row, column);
+                        break;
+                    case "text":
+                        doSpellcheck && parseRowForSpellcheck.call(this, token.value, row, column);
+                        this.myUnfinishedCommentParsing = false;
                         break;
                     default:
-                        // Skip tokens we don't want to spellcheck
+                        // Skip tokens we don't want to parse
+                        this.myUnfinishedCommentParsing = false;
                         break;
                 }
             }
         }
+
+        function parseRowForSpellcheck(tokenValue, row, column) {
+            var words = this.splitTextToWords(tokenValue, row, column);
+            for (var i = 0, wordsLength = words.length; i < wordsLength; ++i) {
+                if (!this.myDictionary.check(words[i].value)) {
+                    this.myErrors.push({
+                        row: words[i].row,
+                        column: words[i].column,
+                        text: "grammar error",
+                        type: "error",
+                        raw: words[i].value
+                    });
+                }
+            }
+        };
+
+        function parseRowForTextComments(tokenValue, row) {
+            // comment format: % TYPE <EMAIL> DATETIME:
+            // {
+            //     TYPE: '*' for the first comment in discussion, '^' for replies
+            //     EMAIL: any string
+            //     DATETIME: any string
+            // }
+            var matchComment = tokenValue.match(COMMENT_REGEXP);
+            if (matchComment) {
+                this.myComments.push({
+                    type: matchComment[1],
+                    email: matchComment[2],
+                    datetime: matchComment[3],
+                    row: row,
+                    text: []
+                });
+                this.myUnfinishedCommentParsing = true;
+            } else if (this.myUnfinishedCommentParsing) {
```

Comment:
It doesn't make too much difference, but you can keep the last created comment object in a variable instead of the boolean flag. If it is null then you have no last comment, if it is not null then you append text here